test-interface/src/main/scala/scala/scalanative/testinterface/TestMainBase.scala

```diff
@@ -117,55 +80,46 @@ abstract class TestMainBase {
         val taskInfos = newTasks.zipWithIndex.map {
           case (t, id) => task2TaskInfo(id + origSize, t, runner)
         }
-        send(client_socket)(TaskInfos(taskInfos))
-        testRunner(tasks ++ newTasks, runner, client_socket)
+        send(clientSocket)(TaskInfos(taskInfos))
+        testRunner(tasks ++ newTasks, runner, clientSocket)
 
       case Command.RunnerDone(_) =>
         val r       = runner.done()
         val message = Command.RunnerDone(r)
-        send(client_socket)(message)
+        send(clientSocket)(message)
 
       case other =>
         println(s"Unexpected message: $other")
     }
+  }
 
   private def task2TaskInfo(id: Int, task: Task, runner: Runner) =
     TaskInfo(id, task.taskDef, task.tags)
 
-  /** Reads `len` bytes from `client` socket. */
-  private def read(client: CInt)(len: Int): DataInputStream = {
-    val buf = new Array[Byte](len)
-    val _   = socket.recv(client, buf.asInstanceOf[ByteArray].at(0), len, 0)
-    new DataInputStream(new ByteArrayInputStream(buf))
-  }
-
   /** Receives a message from `client`. */
-  private def receive[T](client: CInt): Message = {
-    val msglen = read(client)(4).readInt()
-    val in     = read(client)(msglen)
+  private def receive[T](dis: DataInputStream): Message = {
+    val msglen = dis.readInt()
     val msgbuf = new Array[Byte](msglen)
-    var total  = 0
-    while (total < msglen) {
-      total += in.read(msgbuf, total, msglen - total)
+    if (msglen > 0) {
+      dis.readFully(msgbuf)
```

Comment:
The number of bytes read by `dis.readFully` is the length of the buffer, which in this case is `msglen`: 
``` scala
val msgbuf = new Array[Byte](msglen)
```